src/MSALWrapper/GlobalLock.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// GlobalLock is used for preventing code from entering critical sections in different processes at the same time.
+    /// </summary>
+    public class GlobalLock : IDisposable
+    {
+        private Mutex mutex;
+        private bool disposedValue;
+        private bool lockAcquired;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlobalLock"/> class.
+        /// </summary>
+        /// <param name="lockName">Unique lock ID.</param>
+        /// <exception cref="ArgumentNullException">The <see cref="ArgumentNullException"/>.</exception>
+        public GlobalLock(string lockName)
+        {
+            if (lockName == null)
+            {
+                throw new ArgumentNullException($"{nameof(lockName)} must not be null");
+            }
+
+            this.mutex = new Mutex(false, lockName);

```

Comment:
Similar to the notion of "no magic numbers", I also like to encourage no "magic bools". For anyone reading `new Mutex(false, lockName);` it is not clear what the value of `false` is. Prefer extracting this into a named `const` value.